src/doc/rgw/user-accounts.md

```diff
@@ -0,0 +1,73 @@
+# iam accounts
+
+add a user account primitive for use with IAM REST APIs. this enables end-users to use tools like `aws iam create-user` for self-service management of the users, roles, policies, and access keys in their account without requiring global admin privileges for radosgw-admin commands or radosgw /admin REST APIs
+
+add new IAM APIs for User and AccessKey operations. some Role/RolePolicy/UserPolicy operations are already supported, but require special admin capabilities. once a User or Role are added to an account, that account's root user will be granted full access to them
+
+existing users and roles must be added to an account manually, either by a radosgw-admin command or /admin API, before they can be managed by an account root user
+
+
+## namespacing
+
+rgw supports namespace isolation of users and buckets using the 'tenant' name, such that two tenants A and B can both have buckets named 'example-bucket', or users named 'example-user', that are completely indepedent of one another
```

Comment:
@cbodley one question here, just to verify if I'm getting this right, we don't need to rely on tenants to be able to have the same user name on different accounts right?,for example: if you have two accounts, you can have users with the same name in each, but there is a single bucket namespace for everyone(both accounts will belong to the default tenant).

Reply:
essentially yes, account A can have a `UserName=foo` and account B can have a different user with `UserName=foo`. if account A and B belong to the same tenant, their users will share a common bucket namespace

however, the iam User APIs map `UserName` to `RGWUserInfo::display_name` instead of `RGWUserInfo::user_id`. those variables correspond to the `--display-name` and `--uid` arguments to `radosgw-admin user create`

it's the `user_id`s that we require to be unique within the tenant, and that still holds for account users - we just generate a random uuid like '550e8400-e29b-41d4-a716-446655440000' for unique user_ids

user ARNs also look different for account users: `arn:aws:iam::{account_id}:user/{display_name}` vs `arn:aws:iam::{tenant}:user/{user_id}`

so for a user in account RGW12345678901234567, an iam GetUser response would look something like this:
```
    <User>
      <UserId>550e8400-e29b-41d4-a716-446655440000</UserId>
      <Path>/</Path>
      <UserName>foo</UserName>
      <Arn>arn:aws:iam::RGW12345678901234567:user/foo</Arn>
      <CreateDate>2013-10-02T17:01:44Z</CreateDate>
    </User>
```

Reply:
Thanks for the explanation, that is clear now.